src/Tizen.Wearable.CircularUI.Chart.Forms.Renderer/BarChartViewRenderer.cs

```diff
@@ -0,0 +1,752 @@
+                return;
+            }
+
+            if (Element.AxisOption != null)
+            {
+                CaculateAxisSize(canvas);
+            }
+            else
+            {
+                _majorAxisSize = new SKSize(0, 0);
+                _minorAxisSize = new SKSize(0, 0);
+                _referenceItemSize = new SKSize(0, 0);
+            }
+
+            var barSize = CalculateBarSize();
+            GenerateDataTable();
+            var points = CalculatePoints(barSize);
+            if (Element.AxisOption.IsVisibleOfMajorAxisLine)
+            {
+                DrawMajorAxisLine(canvas);
+            }
+
+            if (Element.AxisOption.IsVisibleOfCategoryLabel)
+            {
+                DrawCategoryLabels(canvas, points);
+            }
+
+            if (Element.AxisOption.IsVisibleOfMinorAxisLine)
+            {
+                DrawMinorAxisLine(canvas);
+            }
+
+            if (Element.AxisOption.IsVisibleOfReferenceLabel)
+            {
+                DrawReferenceLabels(canvas, barSize);
+            }
+
+            if (Element.BarBackgroundColorIsVisible)
+            {
+                DrawBarBackground(canvas, points, barSize);
+            }
+
+            if (Element.AxisOption.IsVisibleOfReferenceLine)
+            {
+                DrawReferenceLines(canvas, barSize);
+            }
+
+            DrawBars(canvas, points, barSize);
+
+            if (Element.ValueLabelIsVisible)
+            {
+                DrawValueLabels(canvas, points, barSize);
+            }
+        }
+
+        protected virtual void GenerateDataTable()
+        {
+            if (Element.Data == _prevDataSetArray)
+                return;
+
+            _barChartDataTable = new DataItem[_categoryCount];
+            var items = Element.Data[0].DataItems;
+            for (int i = 0; i < items.Count; i++)
+            {
+                var index = items[i].Key == 0 ? i : items[i].Key - 1;
+                index = index < _categoryCount ? index : _categoryCount - 1;
+                _barChartDataTable[index] = items[i] as DataItem;
+            }
+
+            _prevDataSetArray = Element.Data;
+        }
+
+        protected void CaculateAxisSize(SKCanvas canvas)
+        {
+            if (Element.AxisOption == null)
+                return;
+
+            if (Element.AxisOption.IsVisibleOfMajorAxisLine)
+            {
+                _majorAxisSize.Height = LineSize;
+            }
+
+            if (Element.AxisOption.IsVisibleOfMinorAxisLine)
+            {
+                _minorAxisSize.Width = LineSize;
+            }
+
+            if (Element.AxisOption.IsVisibleOfCategoryLabel)
+            {
+                CalculateCategoryItemSize();
+            }
+
+            if (Element.AxisOption.IsVisibleOfReferenceLabel)
+            {
+                CalculateReferenceItemSize();
+            }
+        }
+
+        private SKRect[] MeasureCategoryLabels()
```

Comment:
How about returing 'IEnumerable<SKRect>'? no need to convert to array
```suggestion
        private IEnumerable<SKRect> MeasureCategoryLabels()
```

Reply:
Done. changed all arrary code in Renderer to 'IEnumerable'.